### DISCUSSION TITLE
Impossible to use the RAG

### DISCUSSION THREAD

**Discussion by Domi31tls** [open-webui/open-webui#4222] [General]
# Bug Report

## Description

When I provide a collection and ask a question, it responds that it doesn't know. I have tried many times, modifying my prompts, the embeddings, etc... nothing happens, the documents don't seem to be loaded.

![image](https://github.com/user-attachments/assets/10fecb8e-1c3c-4b74-b3ee-eef72f429aaa)

**Bug Summary:**
![image](https://github.com/user-attachments/assets/995dc4aa-5d82-410a-9077-3ba13e1560f4)

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Open WebUI Version:** Latest
- **Ollama (if applicable):** latest

- Ubuntu 24.04 

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker

## Additional Information

I test with pdf and text files.

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by sir3mat** [comment]
Did you have enabled the hybrid search? What is the provider chosen to expose the embedding model (sentence transformer or ollama)?

**Comment by justinh-rahb** [comment]
What `num_ctx`/"Context Size" are you running your model?

**Comment by Domi31tls** [comment]
Thank for you help,

I use ollama with chroma/all-minilm-l6-v2-f32. 
![image](https://github.com/user-attachments/assets/b3b6c29e-3bdc-40c9-bba9-2f318d75b491)

**Reply by sir3mat** [reply]
Have you clicked the reset document and database button at the end of the documents page in admin settings?
When you change the embedding model you need to click the reset embedding button

**Reply by oatmealm** [reply]
Yes. I'm aware of that. I reloaded documents and tried different combinations of embedding models. The most serious problem is that it doesn't "see" the attached file, either directly in the chat of from the knowledge base. When it does, the answers are often pure hallucinations as if it's not getting grounded at all. No information what is the problem. I know the documents are being processed because I can see the vectors output to the log for each document.

**Reply by sir3mat** [reply]
Okkk very strange.. uhmm i tried different setup like
1) LLM exposed with vLLM and embedding model exposed with TEI and it works
2) LLM exposed with vLLM and embedding model exposed internally with sentence transformer and it works

So, there is some chance that the issue is related to ollama. Do you have access to the pod or the container that run the ollama server and models? 

You have both the LLM and the Embedding Model exposed with llama?

**Reply by Domi31tls** [reply]
Hello ! 
I'm not using Ollama in Docker, it's installed on a Linux machine equipped with an Nvidia card. The Open Web UI instance is on another machine. However, Open Web UI is under Docker. Under webui, the embedded models are locally stored on the webui instance (see attached screenshot config). I've noticed three things: 
1) If I use text files, it works 'better' 
2) If in the same chat, in the same discussion, I re-upload the files to be analyzed, the system's response is: "I can't find the documents" 
3) The problem is the same with web searches, sometimes it won't do the web search and instead the model will respond. 
4) Open Web UI doesn't follow the conversation flow. It always forgets the previous question and answer.
Before using an interface, I used to write my programs in Python and got better results. I haven't tried other inference yet, but I'm planning to test Xinference with a non-container installation. Any suggestions?"
Thank for your help
Ps: je suis français d'ou antoinelouis/crossencoder-camembert-large-mmarcoFR

![image](https://github.com/user-attachments/assets/35e45c38-dc5f-4303-ad41-4ec6d105fb4d)

I

**Reply by oatmealm** [reply]
Ollama runs natively on silicon and yes both are served by ollama. AFAIK openwebui doesn't yet support ollama's new embedding endpoint...? Maybe that should help?

**Comment by Domi31tls** [comment]
I use thé default num_ctx.

**Comment by justinh-rahb** [comment]
@Domi31tls for Ollama models this is only 2048 tokens, not a lot to work with. Increasing this value will vastly increase RAM usage though. There is a PR in Ollama that would seek to alleviate that burden:
https://github.com/ollama/ollama/pull/5894

**Comment by erickgodev** [comment]
I noticed the same thing using chatgpt it doesn't return any context

**Reply by mbentley** [reply]
Recently I have noticed that when I include a document as context and I ask a question immediately, it has no idea about the context.  If I ask the question again, it suddenly knows.  I honestly have no idea when it started doing this because it used to work fine for me.

**Reply by mbentley** [reply]
Here is an example where the document just has `foo bar baz`

![screenshot 2024-08-01 at 2 57 11 PM](https://github.com/user-attachments/assets/e472e228-c60e-4cec-bc06-dd46ae1ea4be)

**Reply by qiulang** [reply]
My experience is exactly the opposite. If I include a document as the context it can generate the answers correctly. But If I don't specify any document RAG won't work. I have upgraded to latest 0.3.15 and still does not work. You may check my question here https://github.com/open-webui/open-webui/discussions/5002

**Comment by oatmealm** [comment]
Same problem here. I'm realizing  it's practically useless for real RAG project in any meaningful way.

**Comment by PlebeiusGaragicus** [comment]
From what I've seen OI sometimes just crams a part of the document into the system prompt - not in any way what I was hoping - and it doesn't work for some file types.

This feature just hasn't been built-out yet.
